DOC: fix DataFrame.plot docstrings errors PR09,PR08

josibake opened the pull request:
this is related to #28702
also fix a few formatting issues
related to default arguments
and argument ordering

- [x] tests added / passed
- [x] passes `black pandas`

Commit:
fix plotting docstrings errors pr09,pr08

this is related to #28702
also fix a few formatting issues
related to default arguments
and argument ordering

WillAyd reviewed (approved):
lgtm - @datapythonista care to take a look?

datapythonista commented on pandas/plotting/_core.py:
```diff
@@ -507,86 +507,76 @@ class PlotAccessor(PandasObject):
-        - 'pie' : pie plot
-        - 'scatter' : scatter plot
-        - 'hexbin' : hexbin plot
+    kind : {'line', 'bar', 'barh', 'hist', 'box', 'kde', 'density', 'area', \
```
Do you mind reverting this one? We'll change the guidelines, I think the previous is more clear.

josibake replied:
yep, i think this is the same scenario as #28739 , regarding long lists of values

datapythonista commented on pandas/plotting/_core.py:
```diff
@@ -507,86 +507,76 @@ class PlotAccessor(PandasObject):
-    legend : False/True/'reverse'
-        Place legend on axis subplots
+        Axis grid lines.
+    legend : bool or 'reverse'
```
reverse in curly brackets

datapythonista commented on pandas/plotting/_core.py:
```diff
@@ -507,86 +507,76 @@ class PlotAccessor(PandasObject):
     style : list or dict
-        The matplotlib line style per column
+        The matplotlib line style per column.
     logx : bool or 'sym', default False
```
sym in curly brackets too if you don't mind, there are few of those

datapythonista reviewed (commented):
Couple of things, but looks great, thanks for the fixes

Commit:
format for long list of known values, add braces

following the convention for long lists of known values
braces following the convention for multi type arguments

datapythonista reviewed (approved):
looks good, thanks for the fixes @josibake 

Can you just merge master, so we can merge with green CI please?

datapythonista commented:
@josibake do you mind merging master and fixing the conflicts. This is ready to be merged when you do. Thanks!

jbrockmendel commented:
@josibake looks about ready, can you rebase

jbrockmendel commented:
@josibake pls rebase

jbrockmendel commented:
Closing as stale.  Feel free to re-open if you want to continue here.
